--- a/libdnet-stripped/m4/acinclude.m4	2026-08-17 16:26:30.497720595 +0200
+++ b/libdnet-stripped/m4/acinclude.m4	2026-08-17 16:27:42.464285922 +0200
@@ -146,16 +146,18 @@
 dnl (OpenSolaris) may have the PF_PACKET interface. The eth-linux.c code
 dnl activated by this check is specific to Linux.
 AC_DEFUN([AC_DNET_LINUX_PF_PACKET],
-    [AC_MSG_CHECKING(for Linux PF_PACKET sockets)
-    AC_TRY_COMPILE([#include <netpacket/packet.h>
-                    #include <linux/if_ether.h>],
-                   [int foo() { return ETH_P_ALL; }],
-    ac_cv_dnet_linux_pf_packet=yes,
-    ac_cv_dnet_linux_pf_packet=no)
+    [AC_CHECK_DECL([ETH_P_ALL],
+    [ac_cv_dnet_linux_pf_packet=yes],
+    [ac_cv_dnet_linux_pf_packet=no],
+        [
+#include <netpacket/packet.h>
+#include <linux/if_ether.h>
+])
+    AC_MSG_CHECKING(for Linux PF_PACKET sockets)
     AC_MSG_RESULT($ac_cv_dnet_linux_pf_packet)
     if test $ac_cv_dnet_linux_pf_packet = yes ; then
-       AC_DEFINE(HAVE_LINUX_PF_PACKET, 1,
-                 [Define if you have Linux PF_PACKET sockets.])
+    AC_DEFINE(HAVE_LINUX_PF_PACKET, 1,
+          [Define if you have Linux PF_PACKET sockets.])
     fi])
 
 dnl
