diff --git a/BZ2StreamScanner.cpp b/BZ2StreamScanner.cpp
index a134712..4992965 100644
--- a/BZ2StreamScanner.cpp
+++ b/BZ2StreamScanner.cpp
@@ -16,6 +16,8 @@
 #include <cstring>
 #include <cstdio>
 
+#define ARRAY_SIZE(array) (sizeof (array) / sizeof *(array))
+
 using namespace std;
 
 namespace pbzip2
@@ -45,12 +47,12 @@ int BZ2StreamScanner::init( int hInFile, size_t inBuffCapacity )
 	CharType bz2header[] = "BZh91AY&SY";
 	// zero-terminated string
 	CharType bz2ZeroHeader[] =
-		{ 'B', 'Z', 'h', '9', 0x17, 0x72, 0x45, 0x38, 0x50, 0x90, 0 };
+		{ 'B', 'Z', 'h', '9', 0x17, 0x72, 0x45, 0x38, 0x50, 0x90 };
 
 	_hInFile = hInFile;
 	_eof = false;
-	_bz2Header = bz2header;
-	_bz2HeaderZero = bz2ZeroHeader;
+	_bz2Header.assign(bz2header, bz2header + ARRAY_SIZE(bz2header) - 1);
+	_bz2HeaderZero.assign(bz2ZeroHeader, bz2ZeroHeader + ARRAY_SIZE(bz2ZeroHeader));
 	_bz2HeaderFound = false;
 	_inBuffCapacity = 0;
 	_errState = 0;
@@ -361,7 +363,7 @@ BZ2StreamScanner::CharType * BZ2StreamScanner::locateHeaderPrefixInBuff()
 			_errState |= ERR_INVALID_FILE_FORMAT;
 			_inBuffSearchPtr = getInBuffEnd();
 		}
-		else if ( _bz2Header.compare( 0, prefixLen, getInBuffSearchPtr(), prefixLen ) == 0 )
+		else if ( std::equal(_bz2Header.begin(), _bz2Header.begin() + prefixLen, getInBuffSearchPtr()) )
 		{
 			// header prefix found
 		}
@@ -416,7 +418,7 @@ BZ2StreamScanner::CharType * BZ2StreamScanner::searchNextHeaderInBuff()
 	while ( !failed() && ( getUnsearchedCount() >= getHeaderSize() ) )
 	{
 		// _inBuffSearchPtr += prefixLen;
-		basic_string<CharType> * pHdr = NULL;
+		vector<CharType> * pHdr = NULL;
 
 		if ( getInBuffSearchPtr()[hsp] == _bz2Header[hsp] )
 		{
@@ -441,13 +443,13 @@ BZ2StreamScanner::CharType * BZ2StreamScanner::searchNextHeaderInBuff()
 				(*pHdr)[prefixLen] = bwtSizeChar;
 
 				// compare the remaining part of magic header
-				int cmpres = pHdr->compare( hsp, pHdr->size() - hsp,
-						getInBuffSearchPtr() + hsp, pHdr->size() - hsp );
+				bool cmpres = std::equal( pHdr->begin() + hsp, pHdr->end(),
+                                                          getInBuffSearchPtr() + hsp );
 
 				#ifdef PBZIP_DEBUG
 				fprintf( stderr, "   searchNextHeaderInBuff:cmpres=%d\n", cmpres );
 				#endif
-				if ( cmpres == 0 )
+				if ( cmpres )
 				{
 					_searchStatus = true;
 					#ifdef PBZIP_DEBUG
diff --git a/BZ2StreamScanner.h b/BZ2StreamScanner.h
index d3729fe..32697cb 100644
--- a/BZ2StreamScanner.h
+++ b/BZ2StreamScanner.h
@@ -44,7 +44,7 @@ public:
 
 	size_t getInBuffSize() const { return ( _inBuffEnd - _inBuff ); }
 	size_t getInBuffCapacity() const { return _inBuffCapacity; }
-	const basic_string<CharType> & getHeader() const { return _bz2Header; }
+	const vector<CharType> & getHeader() const { return _bz2Header; }
 	size_t getHeaderSize() const { return _bz2Header.size(); }
 	int getErrState() const { return _errState; }
 	bool failed() { return ( _errState != 0 ); }
@@ -125,8 +125,8 @@ private:
 	int _hInFile; // input file descriptor
 	bool _eof;
 
-	basic_string<CharType> _bz2Header;
-	basic_string<CharType> _bz2HeaderZero;
+	vector<CharType> _bz2Header;
+	vector<CharType> _bz2HeaderZero;
 	bool _bz2HeaderFound;
 	bool _searchStatus;
 
