--- a/ssh/operation.vala	2023-03-06 12:28:08.786598670 +0100
+++ b/ssh/operation.vala	2023-03-06 12:30:48.499921573 +0100
@@ -49,7 +49,7 @@
     protected async string? operation_async(string command,
                                             string? input,
                                             Cancellable? cancellable) throws GLib.Error {
-        return_val_if_fail (command != null && command != "", null);
+        return_val_if_fail (command != null && command != "", false);
 
         // Strip the command name for logging purposes
         string[] args;
--- a/ssh/key.vala	2023-03-06 12:31:23.819919257 +0100
+++ b/ssh/key.vala	2023-03-06 12:32:35.643247893 +0100
@@ -230,7 +230,7 @@
      */
     public static async KeyParseResult parse(string data,
                                              Cancellable? cancellable = null) throws GLib.Error {
-        return_if_fail (data != null || data != "");
+        return_val_if_fail (data != null || data != "", false);
 
         var pubkeys = new GenericArray<KeyData>();
         var seckeys = new GenericArray<SecData>();
