--- a/configure.in	2023-06-14 04:01:47.330972834 +0200
+++ b/configure.in	2023-06-14 04:03:00.959501917 +0200
@@ -1482,6 +1482,7 @@
               dnl Check the gcc version macro.
               [
                   #include <stdio.h>
+		  #include <stdlib.h>
 
                   int main()
                   {
@@ -2439,6 +2440,7 @@
                     [
                         #include <zlib.h>
                         #include <stdio.h>
+			#include <stdlib.h>
 
                         int main()
                         {
@@ -2507,6 +2509,7 @@
                 [
                     #include <png.h>
                     #include <stdio.h>
+		    #include <stdlib.h>
 
                     int main()
                     {
--- a/configure	2023-06-14 04:09:10.588157871 +0200
+++ b/configure	2023-06-14 04:10:06.183557388 +0200
@@ -19585,6 +19585,7 @@
 /* end confdefs.h.  */
 
                   #include <stdio.h>
+		  #include <stdlib.h>
 
                   int main()
                   {
@@ -22299,6 +22300,7 @@
 
                         #include <zlib.h>
                         #include <stdio.h>
+			#include <stdlib.h>
 
                         int main()
                         {
@@ -22421,6 +22423,7 @@
 
                     #include <png.h>
                     #include <stdio.h>
+		    #include <stdlib.h>
 
                     int main()
                     {
