From 5b479a1c3139ecbfcf894ac82a4785cbb264c35d Mon Sep 17 00:00:00 2001
From: "David C. Manuelda" <StormByte@gmail.com>
Date: Fri, 4 Apr 2025 21:07:46 +0200
Subject: [PATCH] No gcc link is really needed

---
 src/boot/efi/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/boot/efi/meson.build b/src/boot/efi/meson.build
index a1bdd58..34b7140 100644
--- a/src/boot/efi/meson.build
+++ b/src/boot/efi/meson.build
@@ -152,7 +152,6 @@ if get_option('mode') == 'developer' and get_option('debug')
 endif
 
 efi_c_ld_args = [
-        '-lgcc',
         '-nostdlib',
         '-static-pie',
         '-Wl,--entry=efi_main',
-- 
2.49.0

