--- a/include/Lucene.h	2023-11-23 13:05:01.165486178 +0100
+++ b/include/Lucene.h	2023-11-23 13:05:22.120595134 +0100
@@ -128,14 +128,14 @@
 };
 
 template <class TYPE>
-struct luceneHash : std::unary_function<TYPE, std::size_t> {
+struct luceneHash {
     std::size_t operator()(const TYPE& type) const {
         return type ? type->hashCode() : 0;
     }
 };
 
 template <class TYPE>
-struct luceneWeakHash : std::unary_function<TYPE, std::size_t> {
+struct luceneWeakHash {
     std::size_t operator()(const TYPE& type) const {
         return type.expired() ? 0 : type.lock()->hashCode();
     }
