From acf31e7ae73e0fff93956ffdfefe6cc6d82a9483 Mon Sep 17 00:00:00 2001
From: David C. Manuelda <StormByte@gmail.com>
Date: Sun, 2 Jun 2024 12:48:12 +0200
Subject: [PATCH] Fixed correct boost namespace

---
 src/common/sorting.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/sorting.h b/src/common/sorting.h
index 4958d75..072a6d5 100644
--- a/src/common/sorting.h
+++ b/src/common/sorting.h
@@ -27,7 +27,7 @@ namespace mtx::sort {
 
 template<  typename Titer
          , typename Tcriterion_maker
-         , typename Tcriterion = typename std::result_of< Tcriterion_maker(typename std::iterator_traits<Titer>::value_type) >::type
+         , typename Tcriterion = typename boost::result_of< Tcriterion_maker(typename std::iterator_traits<Titer>::value_type) >::type
          , typename Tcomparator = std::less<Tcriterion>
          >
 void
-- 
2.45.2

