From 320cb2d66871169f19b3cff3451ce74eb59a4c01 Mon Sep 17 00:00:00 2001
From: "David C. Manuelda" <StormByte@gmail.com>
Date: Mon, 26 Aug 2024 12:36:56 +0200
Subject: [PATCH] Remove prohibited register keyword

---
 source/common/md5.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/common/md5.cpp b/source/common/md5.cpp
index a9042f4..8d35700 100644
--- a/source/common/md5.cpp
+++ b/source/common/md5.cpp
@@ -185,7 +185,7 @@ void MD5Final(MD5Context *ctx, uint8_t *digest)
  */
 void MD5Transform(uint32_t *buf, uint32_t *in)
 {
-    register uint32_t a, b, c, d;
+    uint32_t a, b, c, d;
 
     a = buf[0];
     b = buf[1];
-- 
2.46.0

