From f870c07cc843462364e478596c502874769e47d4 Mon Sep 17 00:00:00 2001
From: Geoffrey Mon <geofbot@gmail.com>
Date: Wed, 24 May 2023 18:44:55 -0500
Subject: [PATCH] Fury: clean up speed value handling

---
 .../KingstonFuryDRAMController.cpp                         | 2 +-
 .../RGBController_KingstonFuryDRAM.cpp                     | 7 +++----
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/Controllers/KingstonFuryDRAMController/KingstonFuryDRAMController.cpp b/Controllers/KingstonFuryDRAMController/KingstonFuryDRAMController.cpp
index 0d42cee87..b1c999348 100644
--- a/Controllers/KingstonFuryDRAMController/KingstonFuryDRAMController.cpp
+++ b/Controllers/KingstonFuryDRAMController/KingstonFuryDRAMController.cpp
@@ -105,7 +105,7 @@ void KingstonFuryDRAMController::SetModeCode(unsigned char val)
         int res = bus->i2c_smbus_write_byte_data(0x60 + slots[idx],
                                                  FURY_REG_MODE,
                                                  val);
-        LOG_DEBUG("[%s] %02X setting mode %d; res=%02X",
+        LOG_DEBUG("[%s] %02X setting mode %02X; res=%02X",
                   FURY_CONTROLLER_NAME, 0x60 + slots[idx], val, res);
     }
     std::this_thread::sleep_for(FURY_DELAY);
diff --git a/Controllers/KingstonFuryDRAMController/RGBController_KingstonFuryDRAM.cpp b/Controllers/KingstonFuryDRAMController/RGBController_KingstonFuryDRAM.cpp
index 9f044ebbb..915a978a4 100644
--- a/Controllers/KingstonFuryDRAMController/RGBController_KingstonFuryDRAM.cpp
+++ b/Controllers/KingstonFuryDRAMController/RGBController_KingstonFuryDRAM.cpp
@@ -45,8 +45,8 @@ RGBController_KingstonFuryDRAM::RGBController_KingstonFuryDRAM(KingstonFuryDRAMC
     Rainbow.value      = FURY_MODE_RAINBOW;
     Rainbow.flags      = MODE_FLAG_HAS_SPEED |
         MODE_FLAG_HAS_BRIGHTNESS | MODE_FLAG_HAS_DIRECTION_UD;
-    Rainbow.speed_min  = 0;
-    Rainbow.speed_max  = 60;
+    Rainbow.speed_min  = 60; // speed values are inverted
+    Rainbow.speed_max  = 0;
     Rainbow.speed      = 25;
     Rainbow.brightness_min = 0;
     Rainbow.brightness_max = 100;
@@ -147,9 +147,8 @@ void RGBController_KingstonFuryDRAM::DeviceUpdateLEDs()
 
     if (modes[active_mode].flags & MODE_FLAG_HAS_SPEED)
     {
-        // higher register value represents slower speed
         controller->SetRegister(FURY_REG_SPEED,
-                                modes[active_mode].speed_max - modes[active_mode].speed);
+                                modes[active_mode].speed);
     }
 
     controller->SetRegister(FURY_REG_BRIGHTNESS,
-- 
GitLab

