--- a/autoconf/acspecific.m4	2023-06-14 09:17:59.363954218 +0200
+++ b/autoconf/acspecific.m4	2023-06-14 09:18:25.891121396 +0200
@@ -204,7 +204,7 @@
 [AC_MSG_CHECKING([whether the C compiler ($CC $CFLAGS $LDFLAGS) works])
 AC_LANG_SAVE
 AC_LANG_C
-AC_TRY_COMPILER([main(){return(0);}], ac_cv_prog_cc_works, ac_cv_prog_cc_cross)
+AC_TRY_COMPILER([int main(){return(0);}], ac_cv_prog_cc_works, ac_cv_prog_cc_cross)
 AC_LANG_RESTORE
 AC_MSG_RESULT($ac_cv_prog_cc_works)
 if test $ac_cv_prog_cc_works = no; then
--- a/autoconf/configure	2023-06-14 09:19:16.840443935 +0200
+++ b/autoconf/configure	2023-06-14 09:19:28.880533054 +0200
@@ -1216,7 +1216,7 @@
 #line 1217 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { eval echo configure:1222: \"$ac_link\" 1>&5; eval $ac_link 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
