--- a/libscg/scsierrs.c	2023-06-14 11:27:57.700265490 +0200
+++ b/libscg/scsierrs.c	2023-06-14 11:28:53.496697316 +0200
@@ -992,6 +992,7 @@
 }
 
 
+int
 main()
 {
 	int	i;
@@ -1012,5 +1013,6 @@
 		js_printf("\n");
 	}
 #endif
+	return 0;
 }
 #endif
--- a/autoconf/configure	2023-06-14 11:35:44.622878790 +0200
+++ b/autoconf/configure	2023-06-14 11:36:33.651258165 +0200
@@ -1216,7 +1216,7 @@
 #line 1217 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { eval echo configure:1222: \"$ac_link\" 1>&5; eval $ac_link 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
