From 067966b5f8f23dd7424b35a94d7fbd44ff67982b Mon Sep 17 00:00:00 2001
From: "David C. Manuelda" <StormByte@gmail.com>
Date: Thu, 8 Aug 2024 02:19:13 +0200
Subject: [PATCH] Fix for invalid c++11-narrowing to make clang compile
 successful

---
 src/proctable.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/proctable.cpp b/src/proctable.cpp
index eff30fb..c6a2056 100644
--- a/src/proctable.cpp
+++ b/src/proctable.cpp
@@ -103,7 +103,7 @@ cb_tree_button_pressed (GtkGestureClick*,
     gtk_tree_selection_select_path (selection, path);
   gtk_tree_path_free (path);
 
-  GdkRectangle rect = { x, y, 1, 1 };
+  GdkRectangle rect = { (gint)x, (gint)y, 1, 1 };
 
   gtk_popover_set_pointing_to (GTK_POPOVER (app->proc_popover_menu), &rect);
   gtk_popover_popup (GTK_POPOVER (app->proc_popover_menu));
-- 
2.45.2

