From f40056c2d9f036329ceb812968a7eabcbadc0805 Mon Sep 17 00:00:00 2001
From: David C. Manuelda <stormbyte@gmail.com>
Date: Mon, 8 Jan 2024 18:03:50 +0100
Subject: [PATCH] Fixed scaling factor (hackish way)

---
 libcscreensaver/cs-screen-x11.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libcscreensaver/cs-screen-x11.c b/libcscreensaver/cs-screen-x11.c
index 068788d..91954bb 100644
--- a/libcscreensaver/cs-screen-x11.c
+++ b/libcscreensaver/cs-screen-x11.c
@@ -159,8 +159,8 @@ apply_scale_factor (CsMonitorInfo *infos,
     {
         infos[i].rect.x /= factor;
         infos[i].rect.y /= factor;
-        infos[i].rect.width /= factor;
-        infos[i].rect.height /= factor;
+        //infos[i].rect.width /= factor;
+        //infos[i].rect.height /= factor;
 
         DEBUG ("Scale factor of %d applied.  Monitor %d is %d,%d %d x %d\n",
                factor,
@@ -915,4 +915,4 @@ cs_screen_get_global_scale (void)
     }
 
     return window_scale;
-}
\ No newline at end of file
+}
-- 
2.43.0

