--- a/avidemux/qt4/xdg_data/org.avidemux.Avidemux.appdata.xml.in	2024-02-29 00:47:45.571592300 +0100
+++ b/avidemux/qt4/xdg_data/org.avidemux.Avidemux.appdata.xml.in	2024-02-29 00:48:14.343750772 +0100
@@ -20,7 +20,7 @@
     </categories>
     <url type="homepage">https://www.avidemux.org/</url>
     <provides>
-        <binary>avidemux3_qt5</binary>
+        <binary>avidemux3_qt6</binary>
     </provides>
     <update_contact>Avidemux team -- avidemux.org/smif/</update_contact>
     <screenshots>
--- a/avidemux/qt4/xdg_data/org.avidemux.Avidemux.desktop	2024-02-29 00:47:52.701631571 +0100
+++ b/avidemux/qt4/xdg_data/org.avidemux.Avidemux.desktop	2024-02-29 00:48:58.913996253 +0100
@@ -2,7 +2,7 @@
 Name=Avidemux
 GenericName=Video Editor
 Comment=Multiplatform video editor
-Exec=avidemux3_qt5 %f
+Exec=avidemux3_qt6 %f
 Icon=org.avidemux.Avidemux
 Terminal=false
 Type=Application
