--- a/cupsfilters/pdftoraster.cxx	2024-01-17 19:16:02.412315569 +0100
+++ b/cupsfilters/pdftoraster.cxx	2024-01-17 19:16:19.561422134 +0100
@@ -2201,7 +2201,7 @@
 #define _GLIBCXX_THROW throw
 #endif
 
-void * operator new(size_t size) _GLIBCXX_THROW (std::bad_alloc)
+void * operator new(size_t size)
 {
   return malloc(size);
 }
@@ -2211,7 +2211,7 @@
   free(p);
 }
 
-void * operator new[](size_t size) _GLIBCXX_THROW (std::bad_alloc)
+void * operator new[](size_t size)
 {
   return malloc(size);
 }
