From 17b5a5d508ded470d4d7d2dc476bc3c2eafb4133 Mon Sep 17 00:00:00 2001
From: "David C. Manuelda" <StormByte@gmail.com>
Date: Sat, 1 Jul 2023 15:56:23 +0200
Subject: [PATCH] Fixed missing includes and definitions

---
 scan/sane/hpaio.c   | 1 +
 scan/sane/orblite.c | 7 +++++++
 2 files changed, 8 insertions(+)

diff --git a/scan/sane/hpaio.c b/scan/sane/hpaio.c
index 1497c27..065ff82 100644
--- a/scan/sane/hpaio.c
+++ b/scan/sane/hpaio.c
@@ -51,6 +51,7 @@
 #include "escl.h"
 #include "io.h"
 #include "orblitei.h"
+#include "orblite.h"
 
 
 #define DEBUG_DECLARE_ONLY
diff --git a/scan/sane/orblite.c b/scan/sane/orblite.c
index ba86640..9f8b8ac 100644
--- a/scan/sane/orblite.c
+++ b/scan/sane/orblite.c
@@ -20,6 +20,7 @@
 #include "orblitei.h"
 #include "orblite.h"//Added New
 #include <math.h>
+#include <dlfcn.h>
 #include "utils.h"
 #include "io.h"
 
@@ -45,6 +46,12 @@ static SANE_String SANE_paper_sizes[MAX_PAPER_SIZES+1];
 
 static struct t_SANE *g_handle = NULL;
 
+# ifdef ORBLITE_DEBUG
+# define _DBG(args...) syslog(LOG_INFO, __FILE__ " " STRINGIZE(__LINE__) ": " args)
+# else
+# define _DBG(args...)
+# endif
+
 SANE_Option_Descriptor	DefaultOrbOptions[] = {
 	/* name, title, desc,
 	 * type, unit, size, cap,
-- 
2.41.0

